pkg/metrics/metrics_test.go

```diff
@@ -315,38 +272,209 @@ func TestWriteMetric(t *testing.T) {
 		t.Run(tc.name, func(t *testing.T) {
 			t.Parallel()
 
+			var saveReq *http.Request
+
+			// Request body is intentionally leaked to allow for inspection in test cases.
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if saveReq != nil {
+					t.Fatal("multiple requests in a single test")
+				}
+				saveReq = r.Clone(context.Background())
+				body, err := io.ReadAll(r.Body)
+				if err != nil {
+					t.Fatalf("error copying body: %s", err.Error())
+				}
+				saveReq.Body = io.NopCloser(bytes.NewBuffer(body))
+				r.Body = io.NopCloser(bytes.NewBuffer(body))
+
+				if !strings.HasSuffix(r.RequestURI, "/sendMetrics") {
+					w.WriteHeader(http.StatusNotFound)
+					fmt.Fprintln(w, http.StatusText(http.StatusNotFound))
+					return
+				}
+
+				if strings.HasSuffix(r.RequestURI, "400/sendMetrics") {
+					w.WriteHeader(http.StatusBadRequest)
+					fmt.Fprintln(w, "bad request")
+					return
+				}
+
+				if strings.HasSuffix(r.RequestURI, "500/sendMetrics") {
+					w.WriteHeader(http.StatusInternalServerError)
+					fmt.Fprintln(w, "internal error")
+					return
+				}
```

Comment:
you've re-created `http.NewServeMux` 😄

Reply:
Sometimes you need to rewrite the entire standard library but worse